zad2/warriors.rb

```diff
@@ -1,12 +1,53 @@
-class Warrior
+class Character
+
+attr_accessor :name
+attr_accessor :level
+
   def initialize(name:, level:)
+  	@level = level
+  	@name = name
   end
 
   def strength
+  	@level + [*1..12].sample
+  end
+
+  def card
+  	"#{@name} (lvl #{@level})"
   end
 end
 
+class Warrior < Character
+	def levelup(opponent_level)
+		@level += opponent_level-@level+1 if @level <=opponent_level
```

Comment:
warunek `@level <=opponent_level` można przerobić na guard clause

https://www.thechrisoshow.com/2009/02/16/using-guard-clauses-in-your-ruby-code/

Reply:
Poprawione w inny sposób. Przy jednej linijce guard clause traci sens.